docs/design/hub-and-spoke/switch-validate/switch-validate.md

```diff
@@ -10,4 +10,208 @@ Here is a FSM diagram that helps to understand main steps of validation switch r
 
 ### For more details about hub&spoke and look into examples please follow this [link](https://github.com/telstra/open-kilda/blob/develop/docs/design/hub-and-spoke/v7/README.md)
 
-[Issue #1551](https://github.com/telstra/open-kilda/issues/1551)
\ No newline at end of file
+[Issue #1551](https://github.com/telstra/open-kilda/issues/1551)
+
+## API : GET
+
+### Validate the rules installed on the switch
+
+`GET v1/switches/{switch_id}/rules/validate`
+
+response payload
+
+```json
+{
+  "excess-rules-hex": ["8000000000000001"],
+  "excess_rules": [-9223372036854776000],
+  "missing-rules-hex": ["800000000000001b"],
+  "missing_rules": [-9223372036854776000],
+  "proper-rules-hex": ["8000000000000001"],
+  "proper_rules": [-9223372036854776000]
+}
+```
+
+### Validate the rules and the meters installed on the switch
+
+`GET v1/switches/{switch_id}/validate`
+
+response payload
+
+```json
+{
+  "groups": {
+    "excess": [
+      <groupInfo>
+    ],
+    "misconfigured": [
+      <groupInfo>
+    ],
+    "missing": [
+      <groupInfo>
+    ],
+    "proper": [
+      <groupInfo>
+    ]
+  },
+  "logical_ports": {
+    "error": "some_error",
+    "excess": [
+      <logicalPortInfo>
+    ],
+    "misconfigured": [
+      <logicalPortInfo>
+    ],
+    "missing": [
+      <logicalPortInfo>
+    ],
+    "proper": [
+      <logicalPortInfo>
+    ]
+  },
+  "meters": {
+    "excess": [
+      <meterInfo>
+    ],
+    "misconfigured": [
+      <meterInfo>
+    ],
+    "missing": [
+      <meterInfo>
+    ],
+    "proper": [
+      <meterInfo>
+    ]
+  },
+  "rules": {
+    "excess": [-9223372036854776000],
+    "excess-hex": ["800000000000001b"],
+    "misconfigured": [10, 11],
+    "misconfigured_details": [
+      {
+        "rule_id": "0x8000000000000002",
+        "expected": {
```

Comment:
expected and actual dicts must contain only different values. It will help operators to understand what is wrong with rule. All actual rules also must be on same level as `rule_id` `expected` `actual`